dataland-email-service/src/test/kotlin/org/dataland/datalandemailservice/services/DataRequestAnsweredEmailFactoryTest.kt

```diff
@@ -0,0 +1,110 @@
+    private val receiverEmail = "user@testemail.com"
+    private val dataType = "sfdr"
+    private val dataTypeDescription = "SFDR"
+    private val creationTimestampAsDate = "07 Mar 2024, 15:03"
+    private val closedInDays = "100 days"
+
+    private fun buildTestEmail(
+        setOptionalProperties: Boolean,
+    ): Email {
+        var properties = mapOf(
+            "companyId" to companyId,
+            "companyName" to companyName,
+            "dataType" to dataType,
+            "reportingPeriod" to reportingPeriod,
+            "creationDate" to creationTimestampAsDate,
+        )
+        if (setOptionalProperties) {
+            properties = properties + mapOf(
+                "closedIn" to closedInDays,
+                "dataTypeDescription" to dataTypeDescription,
+            )
+        }
+
+        val email = DataRequestAnsweredEmailFactory(
+            proxyPrimaryUrl = proxyPrimaryUrl,
+            senderEmail = senderEmail,
+            senderName = senderName,
+        ).buildEmail(
+            receiverEmail = receiverEmail,
+            properties = properties,
+        )
+        return email
+    }
+
+    @Test
+    fun `validate that the output of the data request answered mail is correctly formatted`() {
+        val basicEmail = buildTestEmail(false)
+        val emailWithOptionalProperties = buildTestEmail(true)
+        validateEmailFormat(basicEmail, false)
+        validateEmailFormat(emailWithOptionalProperties, true)
+    }
+
+    private fun validateEmailFormat(email: Email, hasOptionalProperties: Boolean) {
+        assertEmailContactInformationEquals(
+            EmailContact(senderEmail, senderName),
+            setOf(EmailContact(receiverEmail)),
+            emptySet(),
+            email,
+        )
+        assertTrue(email.content.htmlContent.contains(companyName))
+        assertTrue(email.content.htmlContent.contains(dataType))
+        assertTrue(email.content.htmlContent.contains(reportingPeriod))
+        assertTrue(email.content.htmlContent.contains(creationTimestampAsDate))
+        assertTrue(
+            email.content.htmlContent.contains(
+                "href=\"https://$proxyPrimaryUrl/companies/$companyId/frameworks/$dataType\"",
+            ),
+        )
+        if (hasOptionalProperties) {
+            assertTrue(email.content.htmlContent.contains(closedInDays))
+            assertTrue(email.content.htmlContent.contains(dataTypeDescription))
+        } else {
+            assertTrue(email.content.htmlContent.contains("some days"))
+        }
+    }
+
+    @Test
+    fun `validate that the text content of the data request answered mail is correctly formatted`() {
+        val basicEmail = buildTestEmail(false)
+        val emailWithOptionalProperties = buildTestEmail(true)
+        validateEmailText(basicEmail)
+        validateEmailText(emailWithOptionalProperties)
+    }
+
+    private fun validateEmailText(email: Email) {
```

Comment:
Verständnisfrage: An dieser Stelle wird das folgende Figma überprüft?  

![grafik](https://github.com/d-fine/Dataland/assets/53607778/442950fd-be37-4b63-a2f6-33a35527c7d0)

An welcher Stelle werden die folgenden Textinhalte überprüft?
- Reporting Period
- How to proceed + Punkt 1 und 2

Zudem sehe ich in dem Figma kein "Go to your data requests" Text.

Reply:
bezüglich: validateEmailText (renamed zu validateDefaultEmailText):
Das wird der Text gecheckt den man sieht falls das email programm keinen html content anzeigen kann (der ist auch hardgecoded und sollte eigentlich nie angezeigt werden). Ich hab da jetzt "Go to your data requests" zu dem Text auf den button geändert.
Reporting period hatte ich bei diesem validateDefaultEmailText tatsächlich vergessen. -> wird jetzt auch überprüft
Der default email text enthält tatsächlich nur die ganz basic sachen, also kein how to proceed? oder Footer/Header content.

Die eigentliche email (also wie sie auf dem Figma aussieht) wird in "...validate email format..." getestet.
Da hab ich jetzt noch ein paar asserTrues für statische Elemente hinzugefügt.

Allgemein hab ich alle methoden ein bisschen renamed, damit dass jetzt hoffentlich nicht mehr so verwirrend ist.